docs/developers/ui-components.rst

```diff
@@ -0,0 +1,301 @@
+.. _developers_ui_components:
+
+======================
+Creating UI components
+======================
+
+UI components are (reusable) logical building blocks to build user interfaces. Some
+obvious examples are buttons and links. More complex examples are footers with navigation
+elements or a form field input with label, input and validation errors.
+
+Development of (new) UI components in Open Forms is subject to some guidelines and
+rules, with the goal of:
+
+* contribute to good accessibility
+* make it possible to *theme* them, so appearance can be tailored to the organization
+* being reusable in similar but different contexts
+* make it easy to maintain the code
+
+The next sections describe how to approach creating or modifying UI components.
+
+Check for existing community components
+=======================================
+
+The `NL Design System components <https://www.nldesignsystem.nl/componenten/>`_
+catalogue already lists a large number of components that may fit your requirements.
+
+You should always check first if there's anything suitable. This can lead to an easily
+re-used community component, or at a minimum provide resources about how to implement
+such a component.
+
+Make sure to read up on how other organizations use components, which design constraints
+they have taken into consideration and where the variation lies if different community
+implementations exist.
+
+If nothing exists that is readily usable, you can proceed with implementing it yourself,
```

Comment:
Should we mention to consider contributing this to NL DS?

Reply:
I have some ideas about this - I don't think there will always be a case to put it in the NL DS organization/maintainer scope. We might instead at some point create a UI components package for OF components, which is then used by the SDK/renderer and could offer re-usable components in the community.

For now, we stick to serving our own needs though.